src/cryptonote_protocol/levin_notify.cpp

```diff
@@ -643,12 +654,20 @@ namespace levin
         const auto start = std::chrono::steady_clock::now();
         noise_channel& channel = zone_->channels.at(channel_);
 
+        bool can_broadcast = true;
+        zone_->p2p->for_connection(channel.connection, [&](detail::p2p_context& context){
+            can_broadcast = context.can_broadcast_txs_to_peer;
+            return true;
+        });
+
         if (!channel.connection.is_nil())
         {
           epee::byte_slice message = nullptr;
-          if (!channel.active.empty())
+          if (can_broadcast && !channel.active.empty())
```

Comment:
This check is insufficient for what you want - you need to actually remove the connection from the "channel". More specifically, the `update_channels` function needs to be filtering when it selects which connection will linked an active "channel.

Reply:
Can you point out why this is insufficient? Once a connection is marked as unusable for broadcast it shouldn't ever broadcast again.Would you prefer immediately calling update_channels, or waiting for the new epoch? I'm concerned that messing with the connection pool for channels could help the ISP determine when a node broadcast based on noise patterns. If a node suddenly stops sending noise but the ISP didn't detect that a Tor stream was shutdown then the
probability that you just sent a transaction was high.

Reply:
Can you point out why this is insufficient? Once a connection is marked as unusable for broadcast it shouldn't ever broadcast again even though it can still be selected as the connection for a noise channel in the future. I can understand removing the connection from the pool but I'm concerned that it could result in noise patterns that the ISP could use to determine when a transaction was broadcast. If I implemented it by immediately removing connections from the pool and updating channels after broadcast then a node could quickly exhaust all connections and have nothing to send noise on. The ISP could detect that noise stopped but no Tor stream shutdown occurred and deduce that the node just broadcast. Eliminating a connection from the pool at the next epoch would probably be better but I don't know if it solves the issue entirely especially for setups with a low tx proxy count. Would you prefer immediately calling update_channels, or waiting for the new epoch?

Reply:
Ok, I see your point, if noise is enabled, you cannot change anything or you alert observers that something changed (a tx was sent). I was thinking about the noise _disabled_ case unfortunately.

I noticed you made no changes for noise disabled, and I think that makes sense given that it just does a flood.

Reply:
The user will be unable to send txes until the next epoch (and new outgoing connections are "selected"). Is that intended? Maybe, given the requirements of this patch.

The bigger issue is that when a new epoch starts, this current iteration may select a connection where sending is already disabled.

Reply:
Sorry for not commenting this on my first comment. The primary issue is how to select the outgoing peers every epoch - it may want to check status before selecting (or not).